MLIR.cmake

```diff
@@ -34,6 +34,29 @@ message(STATUS "BUILD_SHARED_LIBS       : " ${BUILD_SHARED_LIBS})
 set(LLVM_REQUIRES_EH ON)
 message(STATUS "LLVM_REQUIRES_EH        : " ${LLVM_REQUIRES_EH})
 
+# LLVM_HOST_TRIPLE is not properly exported as part of the llvm config. It is actually
+# needed by some of the exported cmake files, so there's work in progress to add it to the
+# LLVM config.
+# If LLVM_HOST_TRIPLE is not available, try to include GetHostTriple.cmake to get the triple.
+# This file is available when MLIR_DIR points to a build (as opposed to an install) directory.
+# When all else fails, default to an empty string which is the old default behavior of onnx-mlir.
+if (DEFINED LLVM_HOST_TRIPLE)
```

Comment:
In your comment below:

> \>**ONNX_MLIR_DEFAULT_TRIPLE**:STRING
   The default triple for which onnx-mlir will generate code. 

But this does not appear to be the case here. `ONNX_MLIR_HOST_TRIPLE` is reset here regardless of if it was defined before or not. 

My understanding is that this entire section here should be predicated on `ONNX_MLIR_HOST_TRIPLE` being undefined.

Reply:
`ONNX_MLIR_HOST_TRIPLE` is calculated each time cmake is invoked, however, `ONNX_MLIR_DEFAULT_TRIPLE` is not reset if it was defined by the user since it's a cache variable. So if I called cmake with: `-DONNX_MLIR_DEFAULT_TRIPLE=my-triple`, `ONNX_MLIR_DEFAULT_TRIPLE` will have the value of `my-triple` even after `set` is called, because the value I gave it was cached, so `set` will not overwrite it.